src/vulkan/engine_vulkan_debugger.cc

```diff
@@ -608,6 +619,53 @@ class Thread : public debug::Thread {
     }
   }
 
+  void ExpectCallstack(
+      const std::vector<debug::StackFrame>& callstack) override {
+    DEBUGGER_LOG("ExpectCallstack()");
+
+    std::vector<dap::StackFrame> got_stack;
+    if (!client_.Callstack(thread_id_, &got_stack)) {
+      return;
```

Comment:
Should this be an error? We expected a callstack but didn't receive on?

Reply:
Callstack already records the error messages before returning `false`:
https://github.com/google/amber/blob/3bb1eea7f9e212e0bae0b5ae7cc1318ff4537217/src/vulkan/engine_vulkan_debugger.cc#L232-L246